env/fs_posix.cc

```diff
@@ -875,6 +875,18 @@ class PosixFileSystem : public FileSystem {
     return IOStatus::OK();
   }
 
+  IOStatus IsDirectory(const std::string& path, const IOOptions& /*opts*/,
+                       bool* is_dir, IODebugContext* /*dbg*/) override {
+    struct stat sbuf;
+    if (stat(path.c_str(), &sbuf) < 0) {
+      return IOError("While doing stat", path, errno);
```

Comment:
Message could be a little more descriptive to indicate the failure is in ```IsDirectory()```, like "While doing stat on directory".